xwiki-rendering-api/src/main/java/org/xwiki/rendering/internal/parser/reference/GenericLinkReferenceParser.java

```diff
@@ -142,7 +142,7 @@ public ResourceReference parse(String rawReference)
         StringBuilder content = new StringBuilder(rawReference);
         resourceReference = parseInterWikiLinks(content);
         if (resourceReference != null) {
-            return resourceReference;
+            resourceReference=resourceReference;
```

Comment:
* setting resourceReference in resourceReference is definitely going to irritate Sonar :)
* if isInWikiMode() return false then you won't get the same result than before